### ISSUE TITLE
feat: Flag missing context when *http.Request is in scope

### ISSUE THREAD

**Issue by BenJetson** [go-simpler/sloglint#102]
Howdy, sloglint maintainers!

I recently built a new API service and wanted to try out some of the latest improvements to the standard library, including `slog` and better `http.ServeMux` routing. Having this linter for `slog` calls has been extremely useful and has pointed out many instances where I was inconsistent with my usage.

However, I noticed that when `context-only` is set to `scope`, the linter does not warn about missing context for logs in `http.HandlerFunc` because the `context.Context` is not a direct parameter.

I think it would be useful to warn about missing context when an `*http.Request` is in scope as well, since you should be passing the request context to the logging function.

For example:

```go
func handleHelloWorld(w http.ResponseWriter, r *http.Request) {
	// missing context, should be flagged:
	slog.Info("hello, world!")

	// correct:
	slog.InfoContext(r.Context(), "hello, world!") 
}
```

This is a common pattern for Go HTTP servers, so I imagine others may find this helpful as well.
